src/main/scala/firrtl/analyses/InstanceKeyGraph.scala

```diff
@@ -52,8 +85,33 @@ class InstanceKeyGraph(c: ir.Circuit) {
     */
   def findInstancesInHierarchy(module: String): Seq[Seq[InstanceKey]] = {
     val instances = vertices.filter(_.module == module).toSeq
-    instances.flatMap{ i => fullHierarchy.getOrElse(i, Nil) }
+    instances.flatMap{ i => cachedFullHierarchy.getOrElse(i, Nil) }
   }
+
+  /** Given a circuit, returns a map from module name to a map
+    * in turn mapping instances names to corresponding module names
+    */
+  def getChildInstanceMap: mutable.LinkedHashMap[OfModule, mutable.LinkedHashMap[Instance, OfModule]] =
+    mutable.LinkedHashMap(childInstances.map { case (k, v) =>
+      val moduleMap: mutable.LinkedHashMap[Instance, OfModule] = mutable.LinkedHashMap(v.map(_.toTokens):_*)
+      TargetToken.OfModule(k) -> moduleMap
+    }:_*)
```

Comment:
I would rather not return mutable things, now that we have 2.13 support, could we return [`SeqMap`](https://www.scala-lang.org/api/2.13.3/scala/collection/immutable/SeqMap.html) for 2.13 (concretely return `VectorMap`)? This would require specializing for 2.13 and 2.12-, at least on this function

Reply:
I would also like comments from @azidar @seldridge and @albert-magyar since this my proposing the first case of this, but I also think we should do this for things like `Record.elements` in Chisel (well changing that from `ListMap` to `SeqMap` in 2.13)

Reply:
I do not want to specialize things for 2.13 in this PR.

Reply:
Fair, I think we should before we crack 3.4